core/codegen/tests/ui-fail/from_form.rs

```diff
@@ -163,4 +163,38 @@ struct Validate3 {
     first: String,
 }
 
+#[derive(FromForm)]
+struct Default0 {
+    #[field(default = "wrong type")]
+    first: i32,
+}
+
+#[derive(FromForm)]
+struct Default1 {
+    #[field(default = < 3)]
+    first: bool,
+}
+
+#[derive(FromForm)]
+struct Default2 {
+    #[field(default = < 3)]
+    first: bool,
+}
+
+struct NotCopy {}
+
+static NOT_COPY: NotCopy = NotCopy {};
+
+#[derive(FromForm)]
+struct Default3 {
+    #[field(default = NOT_COPY)]
+    first: NotCopy,
+}
```

Comment:
The `stderr` for this test is missing, so surely codegen tests will fail.

But this is an interesting example. This should pass, should it not? Otherwise how can we ever set defaults for something like `String`? It would be good to have more passing test cases.

Reply:
Yes this is a mistake. The test should fail because this type is not `Clone`, which should definitely fail. When the type is clone we may want this to work, but in that case a call to `.clone()` needs to be emitted. I don't know how to detect whether a default is a literal/function call, or when the default is a static that needs to be cloned however